vkcs/dataplatform/resource_cluster/expands.go

```diff
@@ -78,6 +138,102 @@ func ExpandClusterConfigsSettings(ctx context.Context, o basetypes.ListValue) ([
 	return result, nil
 }
 
+func ExpandClusterBackup(ctx context.Context, o basetypes.ObjectValue) (*clusters.ClusterCreateConfigMaintenanceBackup, diag.Diagnostics) {
+	backupObjV, diags := ConfigsMaintenanceBackupType{}.ValueFromObject(ctx, o)
+	if diags.HasError() {
+		return nil, diags
+	}
+
+	backup := backupObjV.(ConfigsMaintenanceBackupValue)
+
+	var result clusters.ClusterCreateConfigMaintenanceBackup
+	var hasValue bool
+
+	if o := backup.Full; !o.IsUnknown() && !o.IsNull() {
+		full, diags := ExpandClusterBackupFull(ctx, o)
+		if diags.HasError() {
+			return nil, diags
+		}
+		result.Full = full
+		hasValue = true
+	}
+
+	if o := backup.Incremental; !o.IsUnknown() && !o.IsNull() {
+		incremental, diags := ExpandClusterBackupIncremental(ctx, o)
+		if diags.HasError() {
+			return nil, diags
+		}
+		result.Incremental = incremental
+		hasValue = true
+	}
+
+	if o := backup.Differential; !o.IsUnknown() && !o.IsNull() {
+		differential, diags := ExpandClusterBackupDifferential(ctx, o)
+		if diags.HasError() {
+			return nil, diags
+		}
+		result.Differential = differential
+		hasValue = true
+	}
+
+	if !hasValue {
```

Comment:
Is there a situation when this is needed? When we don't have full, incremental or differential backups and whole backup object is not null or unknown (we check for this in parent function)